homeassistant/components/workday/binary_sensor.py

```diff
@@ -61,16 +66,25 @@ def valid_country(value: Any) -> str:
         vol.Optional(CONF_WORKDAYS, default=DEFAULT_WORKDAYS): vol.All(
             cv.ensure_list, [vol.In(ALLOWED_DAYS)]
         ),
-        vol.Optional(CONF_ADD_HOLIDAYS): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(CONF_REMOVE_HOLIDAYS): vol.All(cv.ensure_list, [cv.string]),
+        vol.Optional(CONF_ADD_HOLIDAYS, default=[]): vol.All(
+            cv.ensure_list, [cv.string]
+        ),
+        vol.Optional(CONF_REMOVE_HOLIDAYS, default=[]): vol.All(
+            cv.ensure_list, [cv.string]
+        ),
```

Comment:
Yes, this is the behavior of voluptuous. If the key isn't present in the data to validate, the validator will not run since there's no item to validate, if the item is marked as Optional without default. If there's a default, the default value will be passed through the validator before returned.